blazor/common/data-binding/dapper-using-webapp.md

```diff
@@ -0,0 +1,570 @@
+                    return Ok(bugCount);
+                }
+            }
+            catch (Exception ex)
+            {
+                // Log the exception or handle it accordingly
+                return StatusCode(500, "Internal Server Error");
+            }
+        }
+
+
+    [HttpPost]
+
+        public async Task<ActionResult<Bug>> Post ( Bug value )
+        {
+            using (IDbConnection db = new SqlConnection(Configuration.GetConnectionString(BUGTRACKER_DATABASE)))
+            {
+                db.Open();
+                await db.ExecuteAsync("insert into bugs (Summary, BugPriority, Assignee, BugStatus) values (@Summary, @BugPriority, @Assignee, @BugStatus)", value) ;
+            }
+            return Ok(value);
+        }
+
+        [HttpPut]
+        public async Task<ActionResult<Bug>> Put ( Bug value )
+        {
+            using (IDbConnection db = new SqlConnection(Configuration.GetConnectionString(BUGTRACKER_DATABASE)))
+            {
+                db.Open();
+                await db.ExecuteAsync("update bugs set Summary=@Summary, BugPriority=@BugPriority, Assignee=@Assignee, BugStatus=@BugStatus where id=@Id", value);
+            }
+            return Ok(value);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task<ActionResult> Delete ( long id )
+        {
+            using (IDbConnection db = new SqlConnection(Configuration.GetConnectionString(BUGTRACKER_DATABASE)))
+            {
+                db.Open();
+                await db.ExecuteAsync("delete from bugs Where id=@BugId", new { BugId = id });
+            }
+            return NoContent();
+        }
+    }
+}
+
+```
+
+Also, make sure to include the `AddControllers` and `MapControllers` methods in the Program.cs file of your server-side application.
+
+{% tabs %}
+{% highlight c# tabtitle="~/Program.cs" hl_lines="1 2 3 4" %}
+....
+builder.Services.AddControllers();
+....
+app.MapControllers();
+....
+{% endhighlight %}
+{% endtabs %}
+
+4. Create a `Services` folder in the `BlazorWebApp.Shared` project. Inside the `Services` folder, create the ClientServices class. This class will be responsible for interacting with the server-side API to perform operations such as retrieving bugs, inserting a new bug, removing a bug, and updating a bug.
+
+{% highlight c# %}
+
+namespace BlazorWebApp.Shared.Services
+{
+    public class ClientServices
+    {
+        private readonly HttpClient _httpClient;
+
+        public ClientServices ( HttpClient httpClient )
+        {
+            _httpClient = httpClient;
+
+        }
+
+        public async Task<List<Bug>> GetBugs ()
+        {
+            var result = await _httpClient.GetFromJsonAsync<List<Bug>>("https://localhost:7112/api/DataGrid");
+
+            return result;
+        }
+
+
+        public async Task<Bug> InsertBug ( Bug value )
+        {
+            await _httpClient.PostAsJsonAsync<Bug>($"https://localhost:7112/api/DataGrid/", value);
```

Comment:
avoid to give local host link correct this for entire document

Reply:
As per discussion , this document was removed.